website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,490 @@
+
+
+<Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/Environment Variables/custom-schema-env-var.png" title="Custom schema environmental variables target name." />
+
+To allow the database/schema/object name to depend on the current branch, you can use the out of the box `DBT_CLOUD_GIT_BRANCH` environment variable in dbt Cloud [special environment variables](/docs/build/environment-variables#special-environment-variables).
+
+
+## Example use cases
+
+Here are some typical examples we've encountered with dbt users leveraging those 3 macros and different logic. 
+
+:::note
+
+Note that the following examples do not comprehensively cover all the available options. These examples are meant to be templates for you to develop your own behaviors. 
+
+:::
+
+
+
+### 1. Using the custom schema without the concatenation with the target schema in production
+
+
+The most common use case is using the custom schema without concatenating it with the default schema name when in production.
+
+To do so, you can create a new file called `generate_schema_name.sql` under your macros folder with the following code:
+
+```Jinja
+
+{% macro generate_schema_name(custom_schema_name, node) -%}
+
+    {%- set default_schema = target.schema -%}
+    {%- if custom_schema_name is none -%}
+
+        {{ default_schema }}
+
+    {%- elif  env_var('DBT_ENV_TYPE','DEV') == 'PROD' -%}
+        
+        {{ custom_schema_name | trim }}
+
+    {%- else -%}
+
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+```
+
+This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
+
+
+| Context     |Target database| Target schema | Resulting object               |
+|-------------|:-------------:|:-------------:|:------------------------------:|
+| Developer 1 | dev           | dbt_dev1      |dev.dbt_dev1_marketing.my_model  |
+| Developer 2 | dev           | dbt_dev2      |dev.dbt_dev2_marketing.my_model  |
+| CI PR 123   | ci            | dbt_pr_123    |ci.dbt_pr_123_marketing.my_model|
+| CI PR 234   | ci            | dbt_pr_234    |ci.dbt_pr_234_marketing.my_model|
+| Production  | prod          | analytics     |prod.marketing.my_model         |
+
+
+:::note
+
+We added logic to check if the current dbt run is happening in production or not. This is important, and we explain why in the section on [What not to do](/guides/customize-schema-alias?step=3#what-not-to-do).
+
+:::
+
+
+### 2. Static schemas - Adding developer identities to the front of tables instead of the schema
+
+Occasionally, we run into instances where the security posture of the organization prevents developers from creating schemas and all developers have to develop in a single schema.
+
```

Comment:
should we reiterate the instructions and tell the user to add these in their macros folder. Is it a separate file for each macro? i think so yea?

e.g.:
Create a new file called generate_schema_name.sql under your macros folder with the following code:

Reply:
Hiya @mirnawong1 

Apologies for the delay - I was reviewing each comment to make sure I understood before committing the changes. Thanks for your suggestion. I agree with you (especially for new users who might view the page). I'm not too sure what the instructions are so tagging @T-Dunlap and @b-per for further assistance

Kind Regards